recipes/nmos-cpp/all/conanfile.py

```diff
@@ -0,0 +1,240 @@
+import json
+import os
+import re
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.33.0"
+
+class NmosCppConan(ConanFile):
+    name = "nmos-cpp"
+    description = "An NMOS C++ Implementation"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/sony/nmos-cpp"
+    topics = ("NMOS")
+
+    settings = "os", "compiler", "build_type", "arch"
+    # for now, no "shared" option support
+    options = {
+        "fPIC": [True, False],
+        "with_dnssd": ["mdnsresponder", "avahi"],
+    }
+    # "fPIC" is handled automatically by Conan, injecting CMAKE_POSITION_INDEPENDENT_CODE
+    default_options = {
+        "fPIC": True,
+        "with_dnssd": "mdnsresponder",
+    }
+
+    # wrapper CMakeLists.txt to call conan_basic_setup()
+    exports_sources = ["CMakeLists.txt"]
+    # use cmake_find_package_multi and prefer config-file packages
+    generators = "cmake", "cmake_find_package_multi"
+
+    short_paths = True
+
+    _cmake = None
+
+    # for out-of-source build, cf. wrapper CMakeLists.txt
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if self.settings.os == "Macos":
+            del self.options.with_dnssd
+        elif self.settings.os == "Linux":
+            self.options.with_dnssd = "avahi"
+        elif self.settings.os == "Windows":
+            self.options.with_dnssd = "mdnsresponder"
+
+    def requirements(self):
+        # for now, consistent with project's conanfile.txt
+        self.requires("boost/1.76.0")
+        self.requires("cpprestsdk/2.10.18")
+        self.requires("websocketpp/0.8.2")
+        self.requires("openssl/1.1.1k")
+        self.requires("json-schema-validator/2.1.0")
+        if self.options.get_safe("with_dnssd") == "mdnsresponder":
+            self.requires("mdnsresponder/878.200.35")
+            #self.options["mdnsresponder"].with_opt_patches = True
+        elif self.options.get_safe("with_dnssd") == "avahi":
+            self.requires("avahi/0.8")
+
+    def build_requirements(self):
+        self.build_requires("cmake/[>3.17]")
```

Comment:
```suggestion
        self.build_requires("cmake/3.17.5")
```

https://github.com/sony/nmos-cpp/blob/14663d595f71d51136ffcd60dd469197bfcbf17d/Development/CMakeLists.txt#L1

We don't allow version ranges

Alternatively, 3.21.1 would be an option too as it is the latest as @prince-chrismc said. I don't have a strong preference either way